ext/mini_racer_extension/mini_racer_extension.cc

```diff
@@ -1185,6 +1234,31 @@ VALUE rescue_callback(VALUE rdata, VALUE exception) {
     return exception;
 }
 
+static void throw_ruby_error(Isolate* isolate, ContextInfo* context_info, VALUE rb_error) {
+        VALUE message = rb_funcall(rb_error, rb_intern("message"), 0);
+	HandleScope scope { isolate };
+
+	Local<ObjectTemplate> errorTemplate = context_info->isolate_info->ruby_error_template.Get(isolate);
+	Local<Object> errorInstance;
+	Local<Context> context = context_info->context->Get(isolate);
+	if(!errorTemplate->NewInstance(context).ToLocal(&errorInstance))
+		return;
```

Comment:
A few of these returns; will end up producing half-formed js error objects and not throwing an exception in JS when an error happens in Ruby.
We should probably try to throw -at least- the old 'ruby exception' we had.

Reply:
If those guard-`return`s are hit, the v8 internals already threw an exception and we just need to get out of the way to let v8 propagate that through JS.

Yes, the v8 API is a bit wierd like that.

Reply:
Ok. Sounds good.